pkg/time/time_skew_amd64.go

```diff
@@ -150,7 +154,154 @@ func (it *TimeSkew) Inject(pid int) error {
 	return nil
 }
 
-func (it *TimeSkew) Recover(pid int) error {
+func (it *SkewClockGetTime) Recover(pid int) error {
 	zeroSkew := NewTimeSkewWithCustomFakeImage(0, 0, it.clockIDsMask, it.fakeImage)
 	return zeroSkew.Inject(pid)
 }
+
+// timeofdaySkewFakeImage is the filename of fake image after compiling
+const timeOfDaySkewFakeImage = "fake_gettimeofday.o"
+
+// vdsoEntryName is the name of the vDSO entry
+// const vdsoEntryName = "[vdso]"
+
+// getTimeOfDay is the target function would be replaced
+const getTimeOfDay = "gettimeofday"
+
+type SkewGetTimeOfDay struct {
+	deltaSeconds     int64
+	deltaNanoSeconds int64
+	fakeImage        *FakeImage
+}
+
+func NewTimeOfDaySkew(deltaSeconds int64, deltaNanoSeconds int64) (*SkewGetTimeOfDay, error) {
```

Comment:
```suggestion
func NewSkewGetTimeOfDay(deltaSeconds int64, deltaNanoSeconds int64) (*SkewGetTimeOfDay, error) {
```

please use the name of the struct as constructor name.